server.py

```
@@ -258,14 +267,58 @@ def make_svg(self, request):
         )
 
+    def make_piece_svg(self, request):
+        piece_set = select_piece_set(request)
+
+        try:
+            raw_size = request.query.get("size")
+            if raw_size is None:
+                raise aiohttp.web.HTTPBadRequest(reason="size query parameter is required")
+            size = int(raw_size)
+        except ValueError:
+            raise aiohttp.web.HTTPBadRequest(reason="size is not a valid number")
```

Ruff (0.16.1)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)